src/client/Client.cc

```diff
@@ -15842,6 +15842,10 @@ int Client::ll_read(Fh *fh, loff_t off, loff_t len, bufferlist *bl)
     return -CEPHFS_ENOTCONN;
   }
 
+  /* We can't return bytes written larger than INT_MAX, clamp len to that */
+  len = std::min(len, (loff_t)INT_MAX);
```

Comment:
Was moving this intentional? I guess it should be fine to keep it right before `int r = _read(fh, off, len, bl);`.?

Reply:
Technically, yes, but this block doesn't require it to be under the lock.

Reply:
I did think about that -- the problem is this dereferences `fh`.

Reply:
oh `tout(cct) << (uintptr_t)fh << std::endl;`, got it.